Error while checking build/O2/fullCI for 2560915ea349b929349035fb85adfb2cd04f3d2c at 2022-10-12 02:15:
```
## sw/BUILD/o2checkcode-latest/log
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/Common/test/testGPUsortCUDA.cu:22:10: error: 'boost/test/unit_test.hpp' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUTracking/TRDTracking/GPUTRDTracker.cxx:37:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUTracking/Base/GPUReconstruction.cxx:37:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUTracking/Base/GPUReconstructionCPU.cxx:45:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUTracking/display/GPUDisplay.cxx:36:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUTracking/Base/cuda/GPUReconstructionCUDAGenRTC.cu:16:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/cuda/../Shared/Utils.h:26:10: error: 'boost/program_options.hpp' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/cuda/../Shared/Utils.h:26:10: error: 'boost/program_options.hpp' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/Framework/Logger/include/Framework/Logger.h:14:10: error: 'fairlogger/Logger.h' file not found [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:520:12: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1059:69: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1175:5: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1852:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:2276:18: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:4336:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/sm_20_atomic_functions.h:89:39: error: redefinition of 'atomicAdd' [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/Framework/Logger/include/Framework/Logger.h:14:10: error: 'fairlogger/Logger.h' file not found [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:520:12: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1059:69: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1175:5: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1852:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:2276:18: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:4336:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/sm_20_atomic_functions.h:89:39: error: redefinition of 'atomicAdd' [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/hip/../Shared/Utils.h:146:34: error: use of undeclared identifier 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/hip/benchmark.hip.cxx:199:35: error: use of undeclared identifier 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/hip/../Shared/Utils.h:146:34: error: use of undeclared identifier 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:8: error: unknown type name '__host__' [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:27: error: expected ';' after top level declarator [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:33: error: overloaded 'operator+=' must have at least one parameter of class or enumeration type [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:44: error: unknown type name 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:53: error: unknown type name 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:66:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:66:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:67:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:67:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:68:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:68:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:69:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:69:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:85:1: error: unknown type name '__global__' [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:19: error: use of undeclared identifier 'blockIdx' [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:32: error: use of undeclared identifier 'blockDim' [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:45: error: use of undeclared identifier 'threadIdx' [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:78: error: use of undeclared identifier 'blockDim' [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALCalibExtractor.h:36:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALCalibExtractor.h:36:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALCalibExtractor.h:36:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10055-slc8_x86-64/0/Detectors/TOF/calibration/src/TOFChannelCalibrator.cxx:23:10: error: 'omp.h' file not found [clang-diagnostic-error]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/10055/2560915ea349b929349035fb85adfb2cd04f3d2c/build_O2_fullCI/pretty.html).

--- a/Detectors/PHOS/calib/src/PHOSCalibWorkflowLinkDef.h
+++ b/Detectors/PHOS/calib/src/PHOSCalibWorkflowLinkDef.h
@@ -37,5 +37,6 @@
 #pragma link C++ class o2::phos::PHOSL1phaseCalibDevice + ;
 #pragma link C++ class o2::calibration::TimeSlot < o2::phos::PHOSL1phaseSlot> + ;
 #pragma link C++ class o2::calibration::TimeSlotCalibration < o2::phos::Cell, o2::phos::PHOSL1phaseSlot> + ;
+#pragma link C++ class vector < int> + ;
 
 #endif

--- a/Detectors/PHOS/calib/src/PHOSL1phaseCalibDevice.cxx
+++ b/Detectors/PHOS/calib/src/PHOSL1phaseCalibDevice.cxx
@@ -50,7 +50,7 @@ void PHOSL1phaseCalibDevice::endOfStream(o2::framework::EndOfStreamContext& ec)
 
   std::vector<int> l1phase{mCalibrator->getCalibration()};
   LOG(info) << "End of stream reached, sending output to CCDB";
-  // prepare all info to be sent to CCDB
+  //prepare all info to be sent to CCDB
   auto flName = o2::ccdb::CcdbApi::generateFileName("L1phase");
   std::map<std::string, std::string> md;
   o2::ccdb::CcdbObjectInfo info("PHS/Calib/L1phase", "L1phase", flName, md, mRunStartTime - o2::ccdb::CcdbObjectInfo::MINUTE,

--- a/Detectors/PHOS/calib/src/PHOSL1phaseCalibrator.cxx
+++ b/Detectors/PHOS/calib/src/PHOSL1phaseCalibrator.cxx
@@ -233,7 +233,7 @@ void PHOSL1phaseCalibrator::endOfStream()
     } else { // RMS are slimilar, chose closer to zero Mean
       bestB = iMinMean;
     }
-    mL1phase |= (bestB << d);
+    mL1phase |= (bestB << (2 * d));
   }
   LOG(info) << "Calculated L1phase=" << mL1phase;
 }

--- a/Detectors/PHOS/reconstruction/include/PHOSReconstruction/Clusterer.h
+++ b/Detectors/PHOS/reconstruction/include/PHOSReconstruction/Clusterer.h
@@ -21,12 +21,12 @@
 #include "DataFormatsPHOS/MCLabel.h"
 #include "DataFormatsPHOS/TriggerRecord.h"
 #include "SimulationDataFormat/MCTruthContainer.h"
+#include "PHOSBase/Geometry.h"
 
 namespace o2
 {
 namespace phos
 {
-class Geometry;
 
 class Clusterer
 {
@@ -48,6 +48,11 @@ class Clusterer
 
   void setBadMap(const o2::phos::BadChannelsMap* m) { mBadMap = m; }
   void setCalibration(const o2::phos::CalibParams* c) { mCalibParams = c; }
+  void setL1phase(int phase)
+  {
+    mL1phase = phase;
+    mSkipL1phase = false;
+  }
 
  protected:
   // Calibrate energy
@@ -60,13 +65,26 @@ class Clusterer
     }
   }
   // Calibrate time
-  inline float calibrateT(float time, short absId, bool isHighGain)
+  inline float calibrateT(float time, short absId, bool isHighGain, int bc)
   {
+    // L1phase correction
+    float shift = 0;
+    if (!mSkipL1phase) {
+      char relid[3];
+      o2::phos::Geometry::absToRelNumbering(absId, relid);
+      int ddl = (relid[0] - 1) * 4 + (relid[1] - 1) / 16 - 2;
+      int l1 = (mL1phase >> (ddl * 2)) & 3; // extract 2 bits corresponding to this ddl
+      l1 = bc % 4 - l1;
+      if (l1 < 0) {
+        l1 += 4;
+      }
+      shift = l1 * 25.e-9;
+    }
     // Calibrate time
     if (isHighGain) {
-      return time - mCalibParams->getHGTimeCalib(absId);
+      return time - mCalibParams->getHGTimeCalib(absId) - shift;
     } else {
-      return time - mCalibParams->getLGTimeCalib(absId);
+      return time - mCalibParams->getLGTimeCalib(absId) - shift;
     }
   }
   // Test Bad map
@@ -88,6 +106,8 @@ class Clusterer
   bool mProcessMC = false;
   int miCellLabel = 0;
   bool mFullCluOutput = false;               ///< Write output full of reduced (no contributed digits) clusters
+  bool mSkipL1phase = true;                  /// Do not correct for L1 phase
+  int mL1phase = 0;                          /// packed shifts for 14 ddls
   Geometry* mPHOSGeom = nullptr;             ///! PHOS geometry
   const CalibParams* mCalibParams = nullptr; ///! Calibration coefficients, Clusterizer not owner
   const BadChannelsMap* mBadMap = nullptr;   ///! Bad map, Clusterizer not owner

--- a/Detectors/PHOS/reconstruction/src/Clusterer.cxx
+++ b/Detectors/PHOS/reconstruction/src/Clusterer.cxx
@@ -15,7 +15,6 @@
 #include "TDecompBK.h"
 
 #include "PHOSReconstruction/Clusterer.h" // for LOG
-#include "PHOSBase/Geometry.h"
 #include "PHOSBase/PHOSSimParams.h"
 #include "DataFormatsPHOS/Cluster.h"
 #include "DataFormatsPHOS/Digit.h"
@@ -86,7 +85,7 @@ void Clusterer::process(gsl::span<const Digit> digits, gsl::span<const TriggerRe
       }
       float x = 0., z = 0.;
       Geometry::absIdToRelPosInModule(digits[i].getAbsId(), x, z);
-      mCluEl.emplace_back(absId, digitSeed.isHighGain(), energy, calibrateT(digitSeed.getTime(), absId, digitSeed.isHighGain()),
+      mCluEl.emplace_back(absId, digitSeed.isHighGain(), energy, calibrateT(digitSeed.getTime(), absId, digitSeed.isHighGain(), tr.getBCData().bc),
                           x, z, digitSeed.getLabel(), 1.);
     }
     mLastElementInEvent = cluelements.size();
@@ -140,7 +139,7 @@ void Clusterer::processCells(gsl::span<const Cell> cells, gsl::span<const Trigge
       }
       float x = 0., z = 0.;
       Geometry::absIdToRelPosInModule(absId, x, z);
-      mCluEl.emplace_back(absId, c.getHighGain(), energy, calibrateT(c.getTime(), absId, c.getHighGain()),
+      mCluEl.emplace_back(absId, c.getHighGain(), energy, calibrateT(c.getTime(), absId, c.getHighGain(), tr.getBCData().bc),
                           x, z, i, 1.);
     }
     mLastElementInEvent = cluelements.size();

--- a/Detectors/PHOS/workflow/include/PHOSWorkflow/ClusterizerSpec.h
+++ b/Detectors/PHOS/workflow/include/PHOSWorkflow/ClusterizerSpec.h
@@ -40,7 +40,7 @@ class ClusterizerSpec : public framework::Task
 {
  public:
   /// \brief Constructor
-  ClusterizerSpec(bool propagateMC, bool scanDigits, bool outputFullClu, bool defBadMap) : framework::Task(), mPropagateMC(propagateMC), mUseDigits(scanDigits), mFullCluOutput(outputFullClu), mDefBadMap(defBadMap) {}
+  ClusterizerSpec(bool propagateMC, bool scanDigits, bool outputFullClu, bool defBadMap, bool skipL1phase = true) : framework::Task(), mPropagateMC(propagateMC), mUseDigits(scanDigits), mFullCluOutput(outputFullClu), mDefBadMap(defBadMap), mSkipL1phase(skipL1phase) {}
 
   /// \brief Destructor
   ~ClusterizerSpec() override = default;
@@ -63,10 +63,12 @@ class ClusterizerSpec : public framework::Task
   bool mFullCluOutput = false;      ///< Write full of reduced (no contributed digits) clusters
   bool mHasCalib = false;           ///< Were calibration objects received
   bool mDefBadMap = false;          ///< Use default bad map and calibration or extract from CCDB
+  bool mSkipL1phase = true;         ///< Do not apply L1phase correction
   bool mInitSimParams = true;       ///< read sim params
   o2::phos::Clusterer mClusterizer; ///< Clusterizer object
   std::unique_ptr<CalibParams> mCalibParams;
   std::unique_ptr<BadChannelsMap> mBadMap;
+  int mL1phase;
   std::vector<o2::phos::Cluster> mOutputClusters;
   std::vector<o2::phos::CluElement> mOutputCluElements;
   std::vector<o2::phos::TriggerRecord> mOutputClusterTrigRecs;
@@ -77,7 +79,7 @@ class ClusterizerSpec : public framework::Task
 ///
 /// Refer to ClusterizerSpec::run for input and output specs
 framework::DataProcessorSpec getClusterizerSpec(bool propagateMC, bool fillFullClu, bool defBadMap = false);
-framework::DataProcessorSpec getCellClusterizerSpec(bool propagateMC, bool fillFullClu, bool defBadMap = false);
+framework::DataProcessorSpec getCellClusterizerSpec(bool propagateMC, bool fillFullClu, bool defBadMap = false, bool skipL1phase = true);
 
 } // namespace reco_workflow
 

--- a/Detectors/PHOS/workflow/include/PHOSWorkflow/RecoWorkflow.h
+++ b/Detectors/PHOS/workflow/include/PHOSWorkflow/RecoWorkflow.h
@@ -46,7 +46,8 @@ framework::WorkflowSpec getWorkflow(bool disableRootInp,
                                     std::string const& cfgOutput = "clusters", //
                                     bool fullCluOut = false,
                                     int flpId = 0,
-                                    bool defBadMap = false);
+                                    bool defBadMap = false,
+                                    bool skipL1phase = true);
 } // namespace reco_workflow
 
 } // namespace phos

--- a/Detectors/PHOS/workflow/src/ClusterizerSpec.cxx
+++ b/Detectors/PHOS/workflow/src/ClusterizerSpec.cxx
@@ -56,6 +56,12 @@ void ClusterizerSpec::run(framework::ProcessingContext& ctx)
     std::decay_t<decltype(ctx.inputs().get<o2::phos::CalibParams*>("calib"))> calibPtr{};
     calibPtr = ctx.inputs().get<o2::phos::CalibParams*>("calib");
     mClusterizer.setCalibration(calibPtr.get());
+
+    if (!mSkipL1phase) {
+      auto vec = ctx.inputs().get<std::vector<int>*>("l1phase");
+      mClusterizer.setL1phase((*vec)[0]);
+    }
+
     mHasCalib = true;
   }
   if (mInitSimParams) { // trigger reading sim/rec parameters from CCDB, singleton initiated in Fetcher
@@ -150,11 +156,11 @@ o2::framework::DataProcessorSpec o2::phos::reco_workflow::getClusterizerSpec(boo
   return o2::framework::DataProcessorSpec{"PHOSClusterizerSpec",
                                           inputs,
                                           outputs,
-                                          o2::framework::adaptFromTask<o2::phos::reco_workflow::ClusterizerSpec>(propagateMC, true, fullClu, defBadMap),
+                                          o2::framework::adaptFromTask<o2::phos::reco_workflow::ClusterizerSpec>(propagateMC, true, fullClu, defBadMap, true),
                                           o2::framework::Options{}};
 }
 
-o2::framework::DataProcessorSpec o2::phos::reco_workflow::getCellClusterizerSpec(bool propagateMC, bool fullClu, bool defBadMap)
+o2::framework::DataProcessorSpec o2::phos::reco_workflow::getCellClusterizerSpec(bool propagateMC, bool fullClu, bool defBadMap, bool skipL1phase)
 {
   // Cluaterizer with cell input
   std::vector<o2::framework::InputSpec> inputs;
@@ -164,6 +170,9 @@ o2::framework::DataProcessorSpec o2::phos::reco_workflow::getCellClusterizerSpec
   if (!defBadMap) {
     inputs.emplace_back("badmap", o2::header::gDataOriginPHS, "PHS_BadMap", 0, o2::framework::Lifetime::Condition, o2::framework::ccdbParamSpec("PHS/Calib/BadMap"));
     inputs.emplace_back("calib", o2::header::gDataOriginPHS, "PHS_Calib", 0, o2::framework::Lifetime::Condition, o2::framework::ccdbParamSpec("PHS/Calib/CalibParams"));
+    if (!skipL1phase && !propagateMC) {
+      inputs.emplace_back("l1phase", o2::header::gDataOriginPHS, "PHS_L1phase", 0, o2::framework::Lifetime::Condition, o2::framework::ccdbParamSpec("PHS/Calib/L1phase"));
+    }
   }
   inputs.emplace_back("recoparams", o2::header::gDataOriginPHS, "PHS_RecoParams", 0, o2::framework::Lifetime::Condition, o2::framework::ccdbParamSpec("PHS/Config/RecoParams"));
   if (propagateMC) {
@@ -181,6 +190,6 @@ o2::framework::DataProcessorSpec o2::phos::reco_workflow::getCellClusterizerSpec
   return o2::framework::DataProcessorSpec{"PHOSClusterizerSpec",
                                           inputs,
                                           outputs,
-                                          o2::framework::adaptFromTask<o2::phos::reco_workflow::ClusterizerSpec>(propagateMC, false, fullClu, defBadMap),
+                                          o2::framework::adaptFromTask<o2::phos::reco_workflow::ClusterizerSpec>(propagateMC, false, fullClu, defBadMap, skipL1phase),
                                           o2::framework::Options{}};
 }

--- a/Detectors/PHOS/workflow/src/RecoWorkflow.cxx
+++ b/Detectors/PHOS/workflow/src/RecoWorkflow.cxx
@@ -60,7 +60,8 @@ o2::framework::WorkflowSpec getWorkflow(bool disableRootInp,
                                         std::string const& cfgOutput,
                                         bool fullCluOut,
                                         int flpId,
-                                        bool defBadMap)
+                                        bool defBadMap,
+                                        bool skipL1phase)
 {
   InputType inputType;
 
@@ -93,7 +94,7 @@ o2::framework::WorkflowSpec getWorkflow(bool disableRootInp,
     }
     if (isEnabled(OutputType::Clusters)) {
       specs.emplace_back(o2::phos::reco_workflow::getRawToCellConverterSpec(static_cast<unsigned int>(flpId)));
-      specs.emplace_back(o2::phos::reco_workflow::getCellClusterizerSpec(false, fullCluOut, defBadMap)); // no MC propagation
+      specs.emplace_back(o2::phos::reco_workflow::getCellClusterizerSpec(false, fullCluOut, defBadMap, skipL1phase)); // no MC propagation
       if (!disableRootOut) {
         specs.emplace_back(o2::phos::getClusterWriterSpec(false));
       }
@@ -128,7 +129,7 @@ o2::framework::WorkflowSpec getWorkflow(bool disableRootInp,
     }
     if (isEnabled(OutputType::Clusters)) {
       // add clusterizer
-      specs.emplace_back(o2::phos::reco_workflow::getCellClusterizerSpec(propagateMC, fullCluOut, defBadMap));
+      specs.emplace_back(o2::phos::reco_workflow::getCellClusterizerSpec(propagateMC, fullCluOut, defBadMap, skipL1phase));
       if (!disableRootOut) {
         specs.emplace_back(o2::phos::getClusterWriterSpec(propagateMC));
       }

--- a/Detectors/PHOS/workflow/src/phos-reco-workflow.cxx
+++ b/Detectors/PHOS/workflow/src/phos-reco-workflow.cxx
@@ -53,6 +53,7 @@ void customize(std::vector<o2::framework::ConfigParamSpec>& workflowOptions)
     {"fullclu-output", o2::framework::VariantType::Bool, false, {"compact of full (with contr. digits) clusters output"}},
     {"flpId", o2::framework::VariantType::Int, 0, {"FLP identification: 0,1,..."}},
     {"defbadmap", o2::framework::VariantType::Bool, false, {"Use default bad map and calib instead of CCDB"}},
+    {"disable-l1phase-corr", o2::framework::VariantType::Bool, false, {"do not apply L1 phase correction (e.g. to MC)"}},
     {"configKeyValues", o2::framework::VariantType::String, "", {"Semicolon separated key=value strings ..."}}};
   o2::raw::HBFUtilsInitializer::addConfigOption(options);
   std::swap(workflowOptions, options);
@@ -84,7 +85,8 @@ o2::framework::WorkflowSpec defineDataProcessing(o2::framework::ConfigContext co
                                                  cfgc.options().get<std::string>("output-type"),
                                                  cfgc.options().get<bool>("fullclu-output"),
                                                  cfgc.options().get<int>("flpId"),
-                                                 cfgc.options().get<bool>("defbadmap"));
+                                                 cfgc.options().get<bool>("defbadmap"),
+                                                 cfgc.options().get<bool>("disable-l1phase-corr"));
   // configure dpl timer to inject correct firstTForbit: start from the 1st orbit of TF containing 1st sampled orbit
   o2::raw::HBFUtilsInitializer hbfIni(cfgc, wf);
 

